server/graphql/common/expenses.ts

```diff
@@ -1208,6 +1210,14 @@ const checkCanUseAccountingCategory = (
     throw new ValidationFailed('This accounting category is not allowed for this host');
   } else if (accountingCategory.kind && accountingCategory.kind !== 'EXPENSE') {
     throw new ValidationFailed('This accounting category is not allowed for expenses');
+  } else if (
+    accountingCategory.expensesTypes &&
+    accountingCategory.expensesTypes.length > 0 &&
+    !accountingCategory.expensesTypes.includes(expenseType)
+  ) {
+    throw new ValidationFailed(`This accounting category is not allowed for expenses type: ${expenseType}`);
+  } else if (accountingCategory.hostOnly && !remoteUser?.isAdmin(host.id)) {
+    throw new ValidationFailed('This accounting category can only be used by the host admin');
```

Comment:
I'm a bit worried about these two checks. If a host admin sets the accounting category and then asks the user to edit any info, the system will throw an error saying they're not allowed to use the accounting category.

Possible solutions:
- Do not add any checks; continue to consider `expensesTypes` and `hostOnly` as "visual"/UI limitations
- Only run these two conditions when changing the expense type (`existingExpense.type !== expenseType`)